### PR TITLE
Add path_safe method

### PR DESCRIPTION
**Pull request by bdraco**
Needed for https://github.com/aio-libs/aiohttp/pull/9267

### CODE REVIEW

File: yarl/_url.py [hunk 1]
```diff
@@ -710,6 +711,17 @@ def path(self) -> str:
         """
         return self._PATH_UNQUOTER(self.raw_path)
 
+    @cached_property
+    def safe_path(self) -> str:
```
Comment (bdraco) [inline yarl/_url.py:715]:
```suggestion
    def path_safe(self) -> str:
```

File: yarl/_url.py [hunk 2]
```diff
@@ -225,7 +225,8 @@ class URL:
     _FRAGMENT_REQUOTER = _Quoter(safe="?/:@")
 
     _UNQUOTER = _Unquoter()
-    _PATH_UNQUOTER = _Unquoter(ignore="/", unsafe="+")
+    _PATH_UNQUOTER = _Unquoter(unsafe="+")
+    _SAFE_PATH_UNQUOTER = _Unquoter(ignore="/%", unsafe="+")
```
Comment (bdraco) [inline yarl/_url.py:229]:
```suggestion
    _PATH_SAFE_UNQUOTER = _Unquoter(ignore="/%", unsafe="+")
```

**Comment by Dreamsorcerer** [comment]
Added some docs, let me know if that looks good to you.

**Review by Dreamsorcerer — APPROVED** [review]

File: docs/api.rst [hunk 3]
```diff
@@ -270,6 +270,23 @@ There are two kinds of properties: *decoded* and *encoded* (with
...
+.. attribute:: URL.path_safe
+
+   Similar to :attr:`URL.path` except it doesn't decode ``%2F`` or ``%25``.
+   This allows to distinguish between path separators (``/``) and encoded
+   slashes (``%2F``).
+
+   Note that ``%25`` is also not decoded to avoid issues with double unquoting
+   of values. e.g. You can unquote the value with
+   ``URL.path_safe.replace("%2F", "/").replace("%25", %")`` to get the same
+   result as :meth:`URL.path_safe`. If the `%25` was unqouted, it would be
```
Comment (Dreamsorcerer) [inline docs/api.rst:288]:
```suggestion
   result as :meth:`URL.path_safe`. If the `%25` was unquoted, it would be
```

File: yarl/_url.py [hunk 4]
```diff
@@ -226,6 +226,7 @@ class URL:
 
     _UNQUOTER = _Unquoter()
     _PATH_UNQUOTER = _Unquoter(ignore="/", unsafe="+")
```
Comment (bdraco) [inline yarl/_url.py:228]: Will revert this in the other PR

Comment (bdraco) [reply yarl/_url.py:228]: https://github.com/aio-libs/yarl/pull/1151

File: docs/api.rst [hunk 5]
```diff
@@ -270,6 +270,23 @@ There are two kinds of properties: *decoded* and *encoded* (with
...
+.. attribute:: URL.path_safe
+
+   Similar to :attr:`URL.path` except it doesn't decode ``%2F`` or ``%25``.
+   This allows to distinguish between path separators (``/``) and encoded
+   slashes (``%2F``).
+
+   Note that ``%25`` is also not decoded to avoid issues with double unquoting
+   of values. e.g. You can unquote the value with
+   ``URL.path_safe.replace("%2F", "/").replace("%25", %")`` to get the same
+   result as :meth:`URL.path_safe`. If the ``%25`` was unquoted, it would be
```
Comment (Dreamsorcerer) [inline docs/api.rst:288]:
```suggestion
   result as :meth:`URL.path`. If the ``%25`` was unquoted, it would be
```

**Pull request merged**